pkg/app/instances/docker.go

```diff
@@ -117,16 +119,56 @@ func (m *DockerInstanceManager) DeleteHost(zone string, _ accounts.User, host st
 		return nil, fmt.Errorf("Failed to remove docker container: %w", err)
 	}
 	return &apiv1.Operation{
-		Name: host,
-		Done: true,
+		Name: "deletehost_" + host,
+		Done: false,
 	}, nil
 }
 
-func (m *DockerInstanceManager) WaitOperation(zone string, _ accounts.User, _ string) (any, error) {
+func (m *DockerInstanceManager) WaitCreateHostOperation(host string) (any, error) {
+	ctx := context.TODO()
+	for {
+		res, err := m.Client.ContainerInspect(ctx, host)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to inspect docker container: %w", err)
+		}
+		if res.State.Running {
+			return &apiv1.HostInstance{
+				Name: host,
+			}, nil
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func (m *DockerInstanceManager) WaitDeleteHostOperation(host string) (any, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 3*time.Minute)
+	defer cancel()
+	resCh, errCh := m.Client.ContainerWait(ctx, host, "")
+	select {
+	case <-ctx.Done():
+		return nil, fmt.Errorf("Timeout while waiting for deleting host")
```

Comment:
use our `errors` package `github.com/google/cloud-android-orchestration/pkg/app/errors` i.e`  `return nil, errors.NewServiceUnavailableError("Wait for operation timed out", nil)`